File: ts-sdk/src/sdk/models/operations/getpodcastepisodes.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetPodcastEpisodesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=username" })
  username?: string;
}


export class GetPodcastEpisodesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetPodcastEpisodesQueryParams;
}


export class GetPodcastEpisodesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata({ elemType: shared.PodcastEpisodeIndex })
  podcastEpisodeIndices?: shared.PodcastEpisodeIndex[];

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getpodcastepisodes.go
package operations

import (
	"github.com/speakeasy-sdks/forem-sdks/go-client-sdk/pkg/models/shared"
)

type GetPodcastEpisodesQueryParams struct {
	Page     *int32  `queryParam:"style=form,explode=true,name=page"`
	PerPage  *int32  `queryParam:"style=form,explode=true,name=per_page"`
	Username *string `queryParam:"style=form,explode=true,name=username"`
}

type GetPodcastEpisodesRequest struct {
	QueryParams GetPodcastEpisodesQueryParams
}

type GetPodcastEpisodesResponse struct {
	ContentType           string
	PodcastEpisodeIndices []shared.PodcastEpisodeIndex
	StatusCode            int64
}
